enterprise/backend/src/metabase_enterprise/metabot_v3/tools/table_view_config.clj

```diff
@@ -0,0 +1,178 @@
+                                                        :additionalProperties false
+                                                        :properties {:field_id {:type "integer"}
+                                                                     :style {:type "string"
+                                                                             :enum ["normal" "bold" "dim" "title"]}}
+                                                        :required ["field_id" "style"]}}}
+                          :required ["id" "title" "direction" "fields"]}}}
+      :required ["sections"]}}
+    :required ["object_view"]}
+
+   :gallery
+   {:type "object"
+    :additionalProperties false
+    :properties
+    {:object_view
+     {:type "object"
+      :additionalProperties false
+      :properties
+      {:sections {:type "array"
+                  :items {:type "object"
+                          :additionalProperties false
+                          :properties {:id {:type "integer"}
+                                       :title {:type "string"}
+                                       :direction {:type "string"
+                                                   :enum ["horizontal" "vertical"]}
+                                       :fields {:type "array"
+                                                :items {:type "object"
+                                                        :additionalProperties false
+                                                        :properties {:field_id {:type "integer"}
+                                                                     :style {:type "string"
+                                                                             :enum ["normal" "bold" "dim" "title"]}}
+                                                        :required ["field_id" "style"]}}}
+                          :required ["id" "title" "direction" "fields"]}}}
+      :required ["sections"]}}
+    :required ["object_view"]}})
+
+(defn- call-openai-api
+  "Call OpenAI API to generate configuration using structured outputs"
+  [prompt view-type]
+  (let [api-key         (get-openai-api-key)
+        response-schema (get response-schemas view-type)
+        response        (http/post "https://api.openai.com/v1/chat/completions"
+                                   {:headers {"Authorization" (str "Bearer " api-key)
+                                              "Content-Type" "application/json"}
+                                    :body (json/generate-string
+                                           {:model "gpt-4o-mini"
+                                            :messages [{:role "system"
+                                                        :content "You are a UI configuration expert. Generate optimal view configurations based on field metadata."}
+                                                       {:role "user"
+                                                        :content prompt}]
```

Comment:
I was surprised that you have such a skinny system prompt, and give so much "meta" guidance in the user instruction. Not sure how much this would affect this task, given our prompt is entirely system generated.

Reply:
I'm guessing there are big cost and latency savings with using a fixed system prompt too with how most models do caching.